src/test/scala/za/co/absa/hyperdrive/trigger/api/rest/utils/JobTemplateResolutionUtilTest.scala

```diff
@@ -170,6 +170,35 @@ class JobTemplateResolutionUtilTest extends FlatSpec with Matchers {
     )
```

Comment:
Hmm is this behavior correct ? It is the same for (JobTemplate and SparkExecutor)
additionalSparkConfig: "spark.driver.extraJavaOptions" -> "-user.prop=userDriver -aaa.bbb=xxx"
templateParameters: "spark.driver.extraJavaOptions" -> "-template.prop=templateDriver -aaa.bbb=yyy"
Result is:
"spark.driver.extraJavaOptions" -> "-template.prop=templateDriver -aaa.bbb=yyy -user.prop=userDriver -aaa.bbb=xxx"

Reply:
Yes, it's the desired behavior. Both the template properties and the user properties should be included in the final extraJavaOptions. If you want to override a template option, the assumption is that for java options, the succeeding options override the preceding options, that's why the user options are after the template options